### PR TITLE
[FLINK-7531] Move Flink legacy rest handler to flink-runtime

### PR DESCRIPTION
**Pull request by tillrohrmann**
## What is the purpose of the change

Moves the existing REST handler from `flink-runtime-web` to `flink-runtime`.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-runtime/src/test/java/org/apache/flink/runtime/rest/RestEndpointITCase.java [hunk 1]
```diff
@@ -72,14 +72,14 @@ public void testEndpoints() throws Exception {
 		RestServerEndpointConfiguration serverConfig = RestServerEndpointConfiguration.fromConfiguration(config);
 		RestClientConfiguration clientConfig = RestClientConfiguration.fromConfiguration(config);
 
-		final String address = "localhost";
```
Comment (zentol) [inline flink-runtime/src/test/java/org/apache/flink/runtime/rest/RestEndpointITCase.java]: it appears this change belongs into another commit.

File: flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/legacy/backpressure/BackPressureStatsTrackerTest.java [hunk 2]
```diff
@@ -80,11 +73,11 @@ public void execute(Runnable runnable) {
 
 		ExecutionVertex[] taskVertices = new ExecutionVertex[4];
 
-		ExecutionJobVertex jobVertex = mock(ExecutionJobVertex.class);
-		when(jobVertex.getJobId()).thenReturn(new JobID());
```
Comment (zentol) [inline flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/legacy/backpressure/BackPressureStatsTrackerTest.java]: these changes are all unnecessary.

File: flink-runtime/src/test/java/org/apache/flink/runtime/rest/RestEndpointITCase.java (hunk 1 above)
Comment (tillrohrmann) [reply flink-runtime/src/test/java/org/apache/flink/runtime/rest/RestEndpointITCase.java]: Sorry, this must be a rebasing artifact. Will remove it.

File: flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/legacy/backpressure/BackPressureStatsTrackerTest.java (hunk 2 above)
Comment (tillrohrmann) [reply flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/legacy/backpressure/BackPressureStatsTrackerTest.java]: I think these changes are the result of the automatic import optimization.

**Comment by tillrohrmann** [comment]
Thanks for the review @zentol. Once the preceding PRs are merged and Travis gives green light, I would like to merge this PR.
